docs/modules/serialization/pages/compact-serialization.adoc

```diff
@@ -35,368 +35,1501 @@ a class to implement a particular interface. Serializers might be implemented an
+|
+| Optional[list[Optional[int]]]
+|
+| Array of nullable int64s or null.
+
+| FLOAT32
+| float
+|
+| float
+|
+| float
+|
+| 32-bit IEEE 754 floating point number.
+
+| ARRAY_OF_FLOAT32
+| float[]
+|
+| boost::optional<std::vector<float>>
+|
+| Optional[list[float]]
+|
+| Array of float32s or null.
+
+| NULLABLE_FLOAT32
+| Float
+|
+| boost::optional<float>
+|
+| Optional[float]
+|
+| A float32 that can also be null.
+
+| ARRAY_OF_NULLABLE_FLOAT32
+| Float[]
+|
+| boost::optional<std::vector<boost::optional<float>>>
+|
+| Optional[list[Optional[float]]]
+|
+| Array of nullable float32s or null.
+
+| FLOAT64
+| double
+|
+| double
+|
+| float
+|
+| 64-bit IEEE 754 floating point number.
+
+| ARRAY_OF_FLOAT64
+| double[]
+|
+| boost::optional<std::vector<double>>
+|
+| Optional[list[float]]
+|
+| Array of float64s or null.
+
+| NULLABLE_FLOAT64
+| Double
+|
+| boost::optional<double>
+|
+| Optional[float]
+|
+| A float64 that can also be null.
+
+| ARRAY_OF_NULLABLE_FLOAT64
+| Double[]
+|
+| boost::optional<std::vector<boost::optional<double>>>
+|
+| Optional[list[Optional[float]]]
+|
+| Array of nullable float64s or null.
+
+| STRING
+| String
+|
+| std::string
+|
+| Optional[str]
+|
+| A UTF-8 encoded string or null.
+
+| ARRAY_OF_STRING
+| String[]
+|
+| boost::optional<std::vector<std::string>>
+|
+| Optional[list[Optional[str]]]
+|
+| Array of strings or null.
+
+| DECIMAL
+| BigDecimal
+|
+| boost::optional<hazelcast::client::big_decimal>
+|
+| Optional[decimal.Decimal]
+|
+| Arbitrary precision and scale floating point number or null.
+
+| ARRAY_OF_DECIMAL
+| BigDecimal[]
+|
+| boost::optional<std::vector<boost::optional<hazelcast::client::big_decimal>>>
+|
+| Optional[list[decimal.Decimal]]
+|
+| Array of decimals or null.
+
+| TIME
+| LocalTime
+|
+| boost::optional<hazelcast::client::local_time>
+|
+| Optional[datetime.time]
+|
+| Time consisting of hours, minutes, seconds, and nanoseconds or null.
+
+| ARRAY_OF_TIME
+| LocalTime[]
+|
+| boost::optional<std::vector<boost::optional<local_time>>>
+|
+| Optional[list[Optional[datetime.time]]]
+|
+| Array of times or null.
+
+| DATE
+| LocalDate
+|
+| boost::optional<hazelcast::client::local_date>
+|
+| Optional[datetime.date]
+|
+| Date consisting of year, month, and day of the month or null.
+
+| ARRAY_OF_DATE
+| LocalDate[]
+|
+| boost::optional<std::vector<boost::optional<local_date>>>
+|
+| Optional[list[Optional[datetime.date]]]
+|
+| Array of dates or null.
+
+| TIMESTAMP
+| LocalDateTime
+|
+| boost::optional<hazelcast::client::local_date_time>
+|
+| Optional[datetime.datetime]
+|
+| Timestamp consisting of year, month, day of the month, hour, minutes, seconds,
+and nanoseconds or null.
+
+| ARRAY_OF_TIMESTAMP
+| LocalDateTime[]
+|
+| boost::optional<std::vector<boost::optional<hazelcast::client::local_date_time>>>
+|
+| Optional[list[Optional[datetime.datetime]]]
+|
+| Array of timestamps or null.
+
+| TIMESTAMP_WITH_TIMEZONE
```

Comment:
@OzanCansel I see that we didn't put anything to this and `ARRAY_OF_TIMESTAMP_WITH_TIMEZONE ` column for C++. Is there a reason for that?